tidy3d/components/data/monitor_data.py

```diff
@@ -2047,13 +2047,105 @@ class DirectivityData(MonitorData):
+    )
+    Htheta: FieldProjectionAngleDataArray = pd.Field(
+        ...,
+        title="Htheta",
+        description="Spatial distribution of theta-component of the magnetic field.",
+    )
+    Hphi: FieldProjectionAngleDataArray = pd.Field(
+        ...,
+        title="Hphi",
+        description="Spatial distribution of phi-component of the magnetic field.",
+    )
+
+    @property
+    def directivity(self):
+        """Directivity in the frequency domain as a function of angles theta and phi.
+        Directivity is a dimensionless quantity defined as the ratio of the radiation
+        intensity in a given direction to the average radiation intensity over all directions."""
+
+        power_theta = 0.5 * np.real(self.Etheta * np.conj(self.Hphi))
+        power_phi = 0.5 * np.real(-self.Ephi * np.conj(self.Htheta))
+        power = power_theta + power_phi
+
+        # Normalize the aligned flux by dividing by (4 * pi * r^2) to adjust the flux for
+        # spherical surface area normalization
+        flux_normed = self.flux / (4 * np.pi * self.monitor.proj_distance**2)
+
+        return power / flux_normed
+
+    @property
+    def axial_ratio(self):
+        """Axial Ratio (AR) in the frequency domain as a function of angles theta and phi.
+        AR is a dimensionless quantity defined as the ratio of the major axis to the minor
+        axis of the polarization ellipse.
+        Note
+        ----
+        The axial ratio computation is based on:
+
+        Balanis, Constantine A., "Antenna Theory: Analysis and Design,"
+        John Wiley & Sons, Chapter 2.12 (2016).
+        """
+
+        # Calculate the terms of the equation
+        E1_abs_squared = np.abs(self.Etheta) ** 2
+        E2_abs_squared = np.abs(self.Ephi) ** 2
+        E1_squared = self.Etheta**2
+        E2_squared = self.Ephi**2
+
+        # Axial ratio calculations based on equations (2-65) to (2-67)
+        # from Balanis, Constantine A., "Antenna Theory: Analysis and Design,"
+        # John Wiley & Sons, 2016. These calculations use complex numbers
+        # directly and are equivalent to the referenced equations.
+        AR_numerator = E1_abs_squared + E2_abs_squared + np.abs(E1_squared + E2_squared)
+        AR_denominator = E1_abs_squared + E2_abs_squared - np.abs(E1_squared + E2_squared)
+
+        inds_zero = AR_numerator == 0
+        axial_ratio_inverse = xr.zeros_like(AR_numerator)
+        # Perform the axial ratio inverse calculation where the numerator is non-zero
+        axial_ratio_inverse = axial_ratio_inverse.where(
+            inds_zero, np.sqrt(np.abs(AR_denominator / AR_numerator))
+        )
+
+        # Cap the axial ratio values at 1 / AXIAL_RATIO_CAP
+        axial_ratio_inverse = axial_ratio_inverse.where(
+            axial_ratio_inverse >= 1 / AXIAL_RATIO_CAP, 1 / AXIAL_RATIO_CAP
+        )
+
+        return 1 / axial_ratio_inverse
+
+    @property
+    def left_polarization(self):
```

Comment:
`left_hand_polarization`
in the comments add parenthetically that it is the counter clockwise component of circular polarization

same for the other polarization